Decision records/06 - We split up large component test files by feature.md

```diff
@@ -0,0 +1,41 @@
+# We split up large component test files by feature
+
+## Date
+
+July 30, 2018
+
+## Contributors
+
+* Chris Sauvé
+* Mallory Allen
+* Tzvi Melamed
+* Ismail Syed
+* Gord Pearson
+* Matt Seccafien
+
+## Summary
+
+When a test file for a component grows too large, it should split into several sibling test files, each covering a feature of the larger component.
+
+## History
+
+In Shopify Web, some of our components have grown extremely large, despite good separation of discrete logical areas into separate subcomponents. These components are usually those that represent a "page", and are often needed because of the need to collocate form data and/ or GraphQL handling.
```

Comment:
TBF this is not always true, lots of tests in product show care about image mutations for example

Reply:
It depends on how you draw the lines, if you consider "managing the GraphQL query" a logical area, it is all correctly collocated in the page components